auth_brute_force/models/res_authentication_attempt.py

```diff
@@ -34,29 +37,52 @@ class ResAuthenticationAttempt(models.Model):
     whitelisted = fields.Boolean(
         compute="_compute_whitelisted",
     )
+    banned = fields.Boolean(
+        compute="_compute_banned"
+    )
+
+    @api.multi
+    @api.depends('remote', 'login')
+    def _compute_banned(self):
+        for item in self:
+            item.banned = not self._trusted(item.remote, item.login)
 
     @api.multi
     @api.depends('remote')
     def _compute_metadata(self):
-        for item in self:
-            url = GEOLOCALISATION_URL.format(item.remote)
+        if util.strtobool(self.env["ir.config_parameter"].sudo().get_param(
+            'auth_brute_force.check_remote', 'True'
+        )):
+            for item in self:
+                url = GEOLOCALISATION_URL.format(item.remote)
+                try:
+                    res = json.loads(urlopen(url, timeout=5).read())
+                except Exception:
+                    _logger.warning(
+                        "Couldn't fetch details from %s",
+                        url,
+                        exc_info=True,
+                    )
+                else:
+                    item.remote_metadata = "\n".join(
+                        '%s: %s' % pair for pair in res.items())
+
+    def check_whitelist(self, ip):
```

Comment:
`@api.model`. Make it private method, we don't want it to be called through RPC.